frontend/src/EditQuestionDialog.jsx

```diff
@@ -0,0 +1,229 @@
+import { useState, useRef, useEffect } from "react";
+import {
+  Button,
+  TextField,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  ListItemIcon,
+  Stack,
+  MenuItem,
+  Box,
+} from "@mui/material";
+import CustomSnackbar from "./CustomSnackbar";
+import axios from "axios";
+import ReactQuill from "react-quill";
+import "react-quill/dist/quill.snow.css";
+import Editor from "./Editor";
+import EditIcon from "@mui/icons-material/Edit";
+
+function EditQuestionDialog({ question, onEdit }) {
+  //   const storedQuestions = JSON.parse(localStorage.getItem("questions"));
+  // const storedQuestions=await axios.get(`http://localhost:8080/question/`)
+  const [questions, setQuestions] = useState([]);
+  //   const questions = storedQuestions !== null ? storedQuestions : [];
+
+  const inputRefs = {
+    title: useRef(null),
+    categories: useRef(null),
+    complexity: useRef(null),
+    description: useRef(null),
+  };
+
+  const [questionData, setQuestionData] = useState({
+    title: "",
+    categories: "",
+    complexity: "",
+    description: "",
+  });
+
+  const complexityLevels = ["EASY", "MEDIUM", "HARD"];
+
+  const [open, setOpen] = useState(false);
+
+  const handleClickOpen = () => {
+    fetchData();
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const [isSnackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarMessage, setsnackbarMessage] = useState("");
+
+  const handleDuplicateQuestion = () => {
+    setsnackbarMessage("Duplicate question detected!");
+    setSnackbarOpen(true);
+  };
+
+  const handleEmptyInputField = () => {
+    setsnackbarMessage("Missing fields detected!");
+    setSnackbarOpen(true);
+  };
+
+  const handleSnackbarClose = (event, reason) => {
+    if (reason == "clickaway") {
+      return;
+    }
+    setSnackbarOpen(false);
+  };
+
+  const handleFieldChange = (evt) => {
+    const { name, value } = evt.target;
+    setQuestionData((prevUserData) => ({
+      ...prevUserData,
+      [name]: value,
+    }));
+  };
+
+  const handleDescriptionChange = (value) => {
+    setQuestionData((prevUserData) => ({
+      ...prevUserData,
+      description: value,
+    }));
+  };
+
+  const handleSubmit = async (evt) => {
```

Comment:
when i click on the 'Save Question' button after 
- making no changes to any of the fields
- change any field other than the question's title 

i get the 'duplicate question' error message. i think this is because it is checking the current title (which has not been updated) against the question's title in the question's repository and throwing an error.

instead, we can patch only the updated fields and throw error if required. we can have 2 states (oldQuestionData and questionData) to compare fields that have been updated (can refer to this PR #19 for implementation).

Reply:
Ok! I changed to field update and took out duplicate question checking, which will be handled by backend when it is implemented.